app/charts/column/overlay-columns.tsx

```diff
@@ -28,7 +30,7 @@ export const InteractionColumns = () => {
       {chartData.map((d, i) => (
         <rect
           key={i}
-          x={xScaleInteraction(getX(d)) as number}
+          x={xScaleInteraction(formatDate(getX(d))) as number}
```

Comment:
@noahonyejese I might be wrong, but I am pretty sure we shouldn't do anything with dates here, `getX` should correctly return an X-position, I think the error might be somewhere else 👀 Column chart can also not be date-based, so then it would also fail.

Also, it looks like hovering is broken [here](https://visualization-tool-git-feat-extend-supported-datasets-ixt1.vercel.app/en/create/new?cube=https://health.ld.admin.ch/fsvo/moderhinke_entwicklung/1&dataSource=Int), see that the tooltip only works on the first bar, and displays in a wrong place 😱

Reply:
getX returns date as 2024-10-20 instead of 2024.10.20

Reply:
Yes, I am not sure if this would work, but you may also change how getX is created for combo charts, so that it also includes `formatDate`. This way you don't need conditional logic, and can remove all of the formatDate calls in other places, I think it would be better 🤔 

You can see e.g. https://github.com/visualize-admin/visualization-tool/blob/e5170fa14a7f3f23d9da55ff5d059d7bce3f2ae0/app/charts/combo/combo-line-column-state-props.ts#L61-L64 – maybe by extracting `getX` from here like

```typescript
  const { getX: _getX, ...bandXVariables } = useBandXVariables(x, {
    dimensionsByIri,
    observations,
  });
  const formatDate = useFormatFullDateAuto();
  const getX = useCallback((d) => formatDate(_getX(d)), [_getX, formatDate]);
```

? There might be other things to change, maybe also in https://github.com/visualize-admin/visualization-tool/blob/e5170fa14a7f3f23d9da55ff5d059d7bce3f2ae0/app/charts/combo/combo-state.ts#L23-L32, but might be worth trying :)

Reply:
In worst case, I guess we could have` chartType === "combo" ? formatDate(getX(d)) : getX(d)`, but it would be good to explore better solutions first 🥹

Reply:
Yes alright I will try to go this direction